UltraStar Play/Assets/Common/Model/Song/Voice.cs

```diff
@@ -1,33 +1,51 @@
 using System;
 using System.Collections.Generic;
+using System.Globalization;
 using UnityEngine;
 
 [Serializable]
 public class Voice : ISerializationCallbackReceiver
 {
+    public static readonly string soloVoiceName = "";
+    public static readonly string firstVoiceName = "P1";
```

Comment:
This is a good question. I was unsure about this. I would expect the file to be always P1 and P2 in the note part. I thought about DUETSINGERP[N] as a renaming that is only relevant to show the name in the UI?

Thus, it would be (in pseudo UltraStar txt format):
```
P1
: 0 0 0 lalala
P2
: 0 0 0 lalala
```
and
```
#DUETSINGERP1:Alice
#DUETSINGERP2:Bob
P1
: 0 0 0 bla
P2
: 0 0 0 bla
```
But I would not expect
```
#DUETSINGERP1:Alice
#DUETSINGERP2:Bob
Alice
: 0 0 0 bla
Bob
: 0 0 0 bla
```
Is this correct?

Reply:
>(side note: There also exists P3 in song.txt files for lyrics that are to be sung by P1 and P2.)

Well that is just bad design in my opinion. In general (for more than two players) I did expect P[N] to be the start of player N. This is not the case? Have there ever been plans to support more than two singers? I am unsure now how to handle this.

Reply:
`Is this correct?` << will have to check that in the depths of USDX code

Reply:
and yes, this is bad design from ~ 12 years ago, which we can't fix by just ignoring it 🤕

Reply:
To be honest, I have _never_ come across

Reply:
I just tested the following case with USDX 1.3.5:
```
#DUETSINGERP1:Alice
#DUETSINGERP2:Bob
Alice
: 0 0 0 bla
Bob
: 0 0 0 bla
```
The song does not work this way. This makes sense. Thus, #DUETSINGERP is indeed only a name to show it in the UI, but it is not relevant for parsing the song data. P1, P2, etc. identify the voice.

Reply:
"USDX 1.3.5" << where did you find that old version? Why are people still using that nowadays?

Reply:
It was the newest version that I found that was actually running on Windows 10. I was not able to run version 2017.8. There is some virus message coming from Windows Defender and then the file gets deleted.

I would be pleased if this does not happen with the upcoming version of USDX. My version 1.3.5 has an old bug when searching, every letter is typed twice. But besides this it is working great!

Reply:
yes, defender sometimes falsely detects the Nsis installer version as "malware". The installer is actually fine, but occasionally the malware-samples based auto-generated detection patterns are so shitty that they tend to detect everything that uses those specific versions of the nsis installer.